src/region.rs

```diff
@@ -329,6 +329,41 @@ impl<S: Write + Seek> Region<S> {
     }
 }
 
+impl<S: Read + Seek> IntoIterator for Region<S> {
+    type Item = <RegionIterator<S> as Iterator>::Item;
+    type IntoIter = RegionIterator<S>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        RegionIterator{ inner: self, current: 0 }
+    }
+}
+
+pub struct RegionIterator<S: Read + Seek> {
+    inner: Region<S>,
+    current: usize
+}
+
+impl<S: Read + Seek> Iterator for RegionIterator<S> {
+    type Item = CompoundTag;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.current == REGION_CHUNKS {
+            return None;
+        }
+
+        let x = self.current % 32;
+        let z = self.current / 32;
+
+        self.current += 1;
+
+        let pos = RegionChunkPosition::new(x as u8, z as u8);
+        match self.inner.read_chunk(pos) {
+            Ok(chunk) => Some(chunk),
+            Err(_) => self.next()
```

Comment:
This should be `None` and can be replaced with `.ok()` https://doc.rust-lang.org/std/result/enum.Result.html#method.ok

Reply:
> This should be `None` and can be replaced with `.ok()` https://doc.rust-lang.org/std/result/enum.Result.html#method.ok

It can't, if there is a gap in the chunks (say some chuks in the middle of the region), we'd stop before yielding all available ones!

Example:
one chunk not found, say at chunk 0;0, and 0;1 exists, we'd yield `None` and thus terminate the entire iteration before yielding any others!

Reply:
As the docs state: 
@ https://doc.rust-lang.org/std/iter/trait.Iterator.html#tymethod.next
> Returns None when iteration is finished.

Reply:
Ah, I got it! Thanks for an explanation. Lack of experience with iterators on my side.